packages/x-data-grid/src/hooks/utils/useGridApiEventHandler.ts

```diff
@@ -117,12 +117,23 @@ export const useGridApiEventHandler = createUseGridApiEventHandler(registryConta
 
 const optionsSubscriberOptions: EventListenerOptions = { isFirst: true };
 
+const registeredEvents = new Set<GridEvents>();
 export function useGridApiOptionHandler<Api extends GridApiCommon, E extends GridEvents>(
   apiRef: React.MutableRefObject<Api>,
   eventName: E,
   handler?: GridEventListener<E>,
 ) {
-  // Validate that only one per event name?
```

Comment:
@romgrk  Does it make sense to remove this comment, which motivates me to add these changes 😅

Reply:
Sure, that's an old comment. If you want you can update this PR directly, I'll re-open & merge.

Reply:
I have pushed a commit in this PR, could you re-open it?